framework/docker/da_node.go

```diff
@@ -128,7 +126,7 @@ func (n *DANode) Start(ctx context.Context, opts ...types.DANodeStartOption) err
 
 // Name of the test node container.
 func (n *node) Name() string {
-	return fmt.Sprintf("%s-%s", n.GetType(), SanitizeContainerName(n.TestName))
+	return fmt.Sprintf("%s-%d-%s", n.GetType(), n.Index, SanitizeContainerName(n.TestName))
```

Comment:
adding the index into the host name means we can now create N instances of a given node type.